.github/workflows/uploadReleaseArtifacts.yml

```diff
@@ -117,42 +117,48 @@ jobs:
+          if [[ ${{ github.event_name }} == 'schedule' ]] || [[ ${{ github.event_name }} == 'workflow_dispatch' ]]; then
+            json=$(echo $json | jq -c '.assert = "ON" | .mode = "relwithdebinfo"')
+          fi
+          echo "out=$json" >> $GITHUB_OUTPUT
+      - name: Add Build Config for CIRCT-full shared libs
+        id: add-build-config-circt-full-shared-libs
+        run: |
+          json='{"name":"CIRCT-full shared libs","install_target":"install","package_name_prefix":"circt-full-shared-libs","mode":"release","assert":"OFF","shared":"ON","stats":"ON"}'
           if [[ ${{ github.event_name }} == 'schedule' ]] || [[ ${{ github.event_name }} == 'workflow_dispatch' ]]; then
             json=$(echo $json | jq -c '.assert = "ON" | .mode = "relwithdebinfo"')
           fi
           echo "out=$json" >> $GITHUB_OUTPUT
       - name: Build JSON Payloads
         id: build-json-payloads
         run: |
-          echo '${{ steps.add-build-options.outputs.out }}' | jq -sc . > options.json
-          echo build_configs=`cat options.json` >> $GITHUB_OUTPUT
+          echo '${{ steps.add-build-config-firtool.outputs.out }}' '${{ steps.add-build-config-circt-full-shared-libs.outputs.out }}' | jq -sc . > build_configs.json
+          echo '${{ steps.add-linux.outputs.out }}' '${{ steps.add-macos.outputs.out }}' '${{ steps.add-windows.outputs.out }}' | jq -sc . > runners.json
 
-          echo "Build Config Matrix" >> $GITHUB_STEP_SUMMARY
-          echo "\`\`\`json" >> $GITHUB_STEP_SUMMARY
-          cat options.json | jq >> $GITHUB_STEP_SUMMARY
-          echo "\`\`\`" >> $GITHUB_STEP_SUMMARY
+          cat runners.json build_configs.json | jq -sc '[combinations | add]' > matrix-raw.json
 
-          echo '${{ steps.add-linux.outputs.out }}' '${{ steps.add-macos.outputs.out }}' '${{ steps.add-windows.outputs.out }}' | jq -sc . > runners.json
-          echo runners=`cat runners.json` >> $GITHUB_OUTPUT
+          # Exclude job, `BUILD_SHARED_LIBS` option is not supported on Windows
+          cat matrix-raw.json | jq -c 'del(.[] | select(.os == "windows" and .shared == "ON"))' > matrix.json
+          echo matrix=`cat matrix.json` >> $GITHUB_OUTPUT
 
-          echo "Runner/Operating System Matrix:" >> $GITHUB_STEP_SUMMARY
+          echo "Generated Matrix" >> $GITHUB_STEP_SUMMARY
           echo "\`\`\`json" >> $GITHUB_STEP_SUMMARY
-          cat runners.json | jq >> $GITHUB_STEP_SUMMARY
+          cat matrix.json | jq >> $GITHUB_STEP_SUMMARY
           echo "\`\`\`" >> $GITHUB_STEP_SUMMARY
 
           case ${{ github.event_name }} in
             workflow_dispatch)
-              ENV='[{"runTests":${{ inputs.runTests }}}]'
+              ENV='{"runTests":${{ inputs.runTests }}}'
             ;;
             schedule)
-              ENV='[{"runTests":false}]'
+              ENV='{"runTests":false}'
             ;;
             *)
-              ENV='[{"runTests":true}]'
+              ENV='{"runTests":true}'
             ;;
```

Comment:
I'm surprised this work as I thought GitHub was complaining about `env` needed to be an array and not an object. 😕 This is clearly working on your branch, though.

Reply:
It needed to be an array before because `env:` was under the `matrix:`. If I moved it under the `job-name:`, GitHub Actions would complain about `env:` and `uses:` conflicting. Finally, I ended up referencing `useTests` directly under the `with` below, so it no longer needs to be an array.